client/state/themes/actions.js

```diff
@@ -194,39 +195,79 @@ export function receiveThemes( data, site, queryParams, responseTime ) {
 	};
 }
 
-export function activate( theme, site, source = 'unknown' ) {
+/**
+ * Returns an action object to be used in signalling that a theme activation
+ * has been triggered
+ *
+ * @param  {String}  themeId Theme to be activated
+ * @param  {Object}  siteId Site used for activation
+ * @return {Object}  Action object
+ */
+export function themeActivation( themeId, siteId ) {
+	return {
+		type: THEME_ACTIVATE_REQUEST,
+		themeId,
+		siteId,
+	};
+}
+
+/**
+ * Returns an action object to be used in signalling that a theme activation
+ * has been successfull
+ *
+ * @param  {Object}  theme Theme received
+ * @param  {Number}  siteId Site used for activation
+ * @return {Object}  Action object
+ */
+export function themeActivated( theme, siteId ) {
```

Comment:
We have `themeActivation`, `themeActivated`, `themeActivationFailed`. I think that they should be called `themeActivationRequestAction`, `themeActivationRequestSuccessAction`, `themeActivationRequestFailureAction` maybe a bit long but exactly describe what is going on.
Then we would have intentionaly `dispatch ( themeSomethingSomethingAction )` basically `dispatch( Action )` - and this is nicely descriptive.
I think that we should not inline this, and there are two reasons:
1. We would loose symmetry - all other actions have actions creators, and introduce exception in pattern how we handle actions
2. We loose ability to name what we are doing - this is my thing where I like self describing code, dislike inlining

I also think that in the end `activateTheme` should not be in actions. Its just a procedure that dispatches actions. 

Lastly I am not very opinionated about this :smiley_cat: so it can go either way and still will be good as long as the procedure is correct.